crates/ibc/src/mock/context.rs

```diff
@@ -683,15 +880,17 @@ impl MockContext {
 
     /// Triggers the advancing of the host chain, by extending the history of blocks (or headers).
     pub fn advance_host_chain_height(&mut self) {
+        self.advance_host_chain_height_with_timestamp(self.host_timestamp().expect("Never fails"))
+    }
+
+    /// Triggers the advancing of the host chain, by extending the history of blocks (or headers).
+    pub fn advance_host_chain_height_with_timestamp(&mut self, timestamp: Timestamp) {
         let latest_block = self.history.last().expect("history cannot be empty");
         let new_block = HostBlock::generate_block(
             self.host_chain_id.clone(),
             self.host_chain_type,
             latest_block.height().increment().revision_height(),
-            latest_block
-                .timestamp()
-                .add(self.block_time)
-                .expect("Never fails"),
+            timestamp,
         );
```

Comment:
Here, we logically should have a single method to advance the host chain, managing both the height and timestamp. This method should not accept the timestamp as an argument to align with the `block_time` field of the `MockContext`. When necessary, advancing by a specific duration it either should be done by mulitple calls or setting the `block_time`` other that its default value.

Reply:
Hmm. Correct me if I am wrong. It can happen in real life that the chain was halted but later restarted again from a halted height. In that particular height, the block time can be arbitrarily long. There is no block production logic to avoid this. I tried to mimic that situation here.

On the other hand, I do agree that if we have a `block_time` parameter, we should follow it. If we must follow this, the other solution is to iterate a lot of blocks without updating the light-client to pass over the `trusting_period`. But in my experience, generating tm light clients is expensive, and I couldn't finish the test quickly with the default `trusting_period` - `64000` secs. Maybe we should reduce the `trusting_period` for this test?

Reply:
> It can happen in real life that the chain was halted but later restarted again from a halted height. In that particular height, the block time can be arbitrarily long. 

This subtle scenario can exist too afaik. Good point of view.
However, given most client expiry would occur during regular operations, I am more inclined towards keeping things consistent the `block_time`, and cutting down the trust period makes more sense to me.